### PR TITLE
Gh032 myriad config

### LINKED ISSUE #32: Add Myriad config script
Adapt Rob's setup for the CS Cluster for Myriad

### PR DESCRIPTION
**Pull request by Rachel-Alcraft**
closes #32

### CODE REVIEW

File: platforms/ucl_myriad/install [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+#venv setup, install dependencies
+source ~/.bashrc
+module load python3
+python3 -m venv ${CATH_VENV_PATH}
+. ${CATH_VENV_PATH}/bin/activate
+
+source ~/.bashrc
+module load python3
+module load java/openjdk-11/11.0.1
+pip install --upgrade pip wheel pytest nextflow
```
Comment (robertvi) [inline platforms/ucl_myriad/install:20]: here the pip nextflow module installs a command "nextflow" into the venv which, when run, automatically installs nextflow for you and replaces itself with actual nextflow, and also ensures nextflow in the path, therefore if you install this pip module you shouldn't need to also curl install nextflow as well

File: platforms/ucl_myriad/install [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+export PATH=${CATH_GCLOUD}/google-cloud-sdk/bin:${PATH}
+#login to gcloud, generates interactive prompt
+gcloud auth login ${CATH_GCLOUD_USER}
+
+cd ${CATH_REPO_PATH}
+pip install -e .
+pytest
+cath-af-cli
+# RUN NEXTFLOW
+curl -s https://get.nextflow.io | bash
```
Comment (robertvi) [inline platforms/ucl_myriad/install:36]: see above comment: either use pip or curl install method for nextflow not both?

File: platforms/ucl_myriad/nextflow.config [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    queueSize = 10
+}
+
+process {    
+    executor = 'sge'
+    beforeScript = "source ${launchDir}/platforms/ucl_myriad/pre.sh"
+    cpus = 1
+    memory = '4 GB'
+    time = '10min'
+    clusterOptions ='-pe smp 4 -l mem=3G,h_vmem=1G,h_rt=1:0:0 -N cath_job'
```
Comment (robertvi) [inline platforms/ucl_myriad/nextflow.config:13]: Probably best to make sure cpus and pe smp values match? nextflow will autogenerate a job name so -N option maybe not required? with SGE I think it doesn't make sense to put mem larger than h_vmem as virtual memory (ram + swap I think) is always >= true memory (ram only)

**Review by robertvi — APPROVED** [review]
Looks good.

File: platforms/ucl_myriad/nextflow.config (hunk 3 above)
Comment (Rachel-Alcraft) [reply platforms/ucl_myriad/nextflow.config:13]: For the cluster options, can I leave these in for testing and tuning? These are the tuned options that the nf-core config ues so may not make sense but can we start with them and change after?

File: platforms/ucl_myriad/install (hunk 1 above)
Comment (Rachel-Alcraft) [reply platforms/ucl_myriad/install:20]: Yes thanks, my oversight, I'll remove the pip install.

File: platforms/ucl_myriad/nextflow.config (hunk 3 above)
Comment (robertvi) [reply platforms/ucl_myriad/nextflow.config:13]: Agreed (as we said when we spoke last week)

**Pull request merged**